src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TalismanListener.java

```diff
@@ -292,6 +303,46 @@ public void onBlockBreak(BlockBreakEvent e) {
         }
     }
 
+    @EventHandler
+    public void onExperienceReceive(PlayerExpChangeEvent e) {
+        if (e.getAmount() > 0 && Talisman.checkFor(e, SlimefunItems.TALISMAN_WISE)) {
+            e.setAmount(e.getAmount() * 2);
+        }
+    }
+
+    @EventHandler
+    public void onPlayerDeath(PlayerDeathEvent e) {
+        Player player = e.getEntity();
+        DamageCause dmgCause = player.getLastDamageCause().getCause();
+
+        if (dmgCause == DamageCause.VOID && Talisman.checkFor(e, SlimefunItems.TALISMAN_RESURRECTED)) {
+            SlimefunPlugin.runSync(() -> {
+                PaperLib.teleportAsync(player, getSafeRespawnLocation(player));
+            }, 2);
```

Comment:
I still think `PlayerRespawnEvent#setRespawnLocation(...)` would be much much more reliable here.

Reply:
I cant use the respawnLocation, because they never actually die

Reply:
Where does that happen? In Talisman.checkFor?
If so then you should **definitely** add a lot of  comments in this method because that isn't apparent from a single look.

Reply:
Where does what happen?
@TheBusyBiscuit 

Basically how it flows right now is:
1) PlayerDeathEvent gets fired (The player dies after all event handlers are done handling it)
2) I do checks to see if the talisman is there and whatnot, potion effects get added, ...
3) I teleport the user to a safe location
4) Then the event is cancelled, causing the user not to die (see image)
![image](https://user-images.githubusercontent.com/35664724/95505504-b6288380-09ae-11eb-870a-f614fbb58007.png)
5) user keeps doing its thing